### DISCUSSION TITLE
How to properly use a callback for eventlistener?

### DISCUSSION THREAD

**Discussion by patrikpatrik** [DanielChappuis/reactphysics3d#303] [Q&A]
Hello, Im sorry for asking this but I'm confused on how to create this custom event listener? I've read the manual several times but still dont understand it.

If I have an object that's falling to hit a plane and want to determine if the collision occurred, how would I create a callback for that? It says to use` const CollisionCallback::CallbackData&` but how exactly do I instantiate this to my capsule object falling and hitting my plane:?

**Comment by DanielChappuis** [comment]
As described [here](https://reactphysics3d.com/usermanual.html#x1-7700015.1) in the manual, you need to create your own class that inherits from the [rp3d::EventListener](https://reactphysics3d.com/documentation/api/html/classreactphysics3d_1_1_event_listener.html#ac7f131ce5251277c31cf884b339bda30) class. Then, you need to override the `onContact()` method to receive a callback when collision occurs. When this method is called, the `callbackData `parameter will contains the pairs of bodies that collided with information about the contacts. Take a look at the code example in the manual.

You can find more information about what is inside the CallbackData [here ](https://reactphysics3d.com/documentation/api/html/classreactphysics3d_1_1_collision_callback_1_1_callback_data.html)in the API reference.

Let me know if something is not clear.

**Reply by patrikpatrik** [reply]
Thank you Daniel. I followed that template on the user manual. It does trigger a response, but it seems to output twice for each contact. Is there a way to only trigger once? For instance here is my code:

```
// test event listener class 
class YourEventListener : public rp3d::EventListener {

	// Override the onContact() method 
	virtual void onContact(const CollisionCallback::CallbackData& callbackData) override {

		//callbackData.getContactPair(0);
		//CollisionCallback::ContactPair madeContact = callbackData.getContactPair(0);
		printf("i made contact!\n"); // <-- always outputs twice on first contact hit
	}
};
```
